flask_oauthlib/provider/oauth2.py

```diff
@@ -408,6 +413,12 @@ def decorated(*args, **kwargs):
             except oauth2.OAuth2Error as e:
                 log.debug('OAuth2Error: %r', e)
                 return redirect(e.in_uri(redirect_uri))
+            except Exception as e:
+                log.warning('Exception caught while processing request, %s.' % e)
+                return redirect(add_params_to_uri(
+                    self.error_uri, {'error': str(e) }
+                ))
```

Comment:
This `except`  is awful, it swallow user's exception, not oauthlib's exception. I think it should be removed.

related issue: https://github.com/lepture/flask-oauthlib/issues/277
@Fleurer @falquaddoomi @lepture